FrEee/Scripts/AI/TestAI/Plan.csx

```diff
@@ -0,0 +1,278 @@
+        CurrentPlans.ShipBuildPlans.RemoveAll(x => x.IsComplete); 
+    }
+
+    
+}
+
+/// <summary>
+/// The method of storing plans between turns for the AI. 
+/// </summary>
+[Serializable]
+public class PlanList
+{
+    public List<ColonizationPlan> ColonizationPlans;
+
+    public List<ShipBuildPlan> ShipBuildPlans; 
+
+    [DoNotSerialize]
+    public IEnumerable<Plan> AllPlans { 
+        get
+        {
+            var plans = new List<Plan>();
+            plans.AddRange(ColonizationPlans);
+            plans.AddRange(ShipBuildPlans);
+            return plans; 
+        } 
+    }
+
+
+    public PlanList()
+    {
+        ColonizationPlans = new List<ColonizationPlan>();
+        ShipBuildPlans = new List<ShipBuildPlan>(); 
+    }
+
+    public void AssignAllIds()
+    {
+        foreach (var p in ColonizationPlans)
+            if (p.PlanId == null)
+                p.AssignId();
+        foreach (var p in ShipBuildPlans)
+            if (p.PlanId == null)
+                p.AssignId();
+    }
+
+    /// <summary>
+    /// gets the chosen plan, or null. 
+    /// </summary>
+    /// <param name="planId"></param>
+    /// <returns></returns>
+    public Plan GetPlan(string planId)
+    {
+        return AllPlans.FirstOrDefault(x => x.PlanId == planId); 
+    }
+}
+
+
+[Serializable]
+public abstract class Plan
+{
+    public string PlanId; 
+
+
+    [DoNotSerialize]
+    public virtual string TypeShorthand { get; }
+
+    public bool IsComplete;
+
+    /// <summary>
+    /// The priority of the plan. Higher is better.  
+    /// </summary>
+    public int Priority; 
+
+
+    public void AssignId()
+    {
+        PlanId = $"{TypeShorthand}-{DateTime.UtcNow.Ticks}-{PlanManager.CurrentPlans.AllPlans.Count()}";
+    }
+}
+
+
+[Serializable]
+public class ColonizationPlan : Plan
+{
+    [DoNotSerialize]
+    public override string TypeShorthand => "C";
+
+    /// <summary>
+    /// The planet intended to colonize. 
+    /// </summary>
+    [DoNotSerialize]
+    public Planet Planet { get {
+            if (PlanetId == 0)
+                return null; return GalaxyReference<Planet>.GetGalaxyReference(PlanetId);  } set { PlanetId = value.ID; } }
+
+    /// <summary>
+    /// The planet that will build the colony ship to colonize this world. 
+    /// </summary>
+    [DoNotSerialize]
+    public Planet SourcePlanet { get
+        {
+            if (SourcePlanetId == 0)
+                return null; return GalaxyReference<Planet>.GetGalaxyReference(SourcePlanetId); } set { SourcePlanetId = value.ID; } }
+
+    [DoNotSerialize]
+    public ShipBuildPlan ShipBuildPlan
+    {
+        get
+        {
+            if (ShipBuildPlanId == null)
+                return null;
+
+            return PlanManager.CurrentPlans.ShipBuildPlans.FirstOrDefault(x => x.PlanId == ShipBuildPlanId); 
+        }
+    }
+
+
+    /// <summary>
+    /// The Id of the planet. 
+    /// </summary>
+    public long PlanetId;
```

Comment:
Hmm, you're manually storing IDs? Do we need a third type of reference, besides GalaxyReference and ModReference?